File: 프로그래머스/unrated/181897. 리스트 자르기/리스트 자르기.py
import itertools

def solution(n, slicer, num_list):
    answer = []
    a = slicer[0]
    b = slicer[1]
    c = slicer[2]
    if n == 1 :
        answer.append(num_list[:b+1])
        answer = list(itertools.chain(*answer))
    elif n == 2:
        answer.append(num_list[a:])
        answer = list(itertools.chain(*answer))
    elif n == 3:
        answer.append(num_list[a:b+1])
        answer = list(itertools.chain(*answer))
    elif n == 4:
        for i in range(a,b+1,c):
            answer.append(num_list[i])
        
    return answer

File: 프로그래머스/unrated/181897. 리스트 자르기/리스트 자르기.js
function solution(n, slicer, num_list) {
    let result = [];
    let a = slicer[0];
    let b = slicer[1];
    let c = slicer[2];
    
    switch(n){
        case 1:
            result = num_list.slice(0,b+1);
            break;
        case 2:
            result = num_list.slice(a,num_list.length);
            break;
        case 3:
            result = num_list.slice(a,b+1);
            break;
        case 4:
            for(let i=a;i<=b;i+=c){
                result.push(num_list[i]);
            }
            break;
        default:
            break;
    }
    
    return result;
}